### PR TITLE
feat(fire): define recursive state contract

### PR DESCRIPTION
**Pull request by monocongo**
## Summary
- records recursive fire state, execution, and Dask-time-chunk decisions in ADR-0006
- prototypes pure-NumPy recurrence with append-state round-trip coverage
- makes shared xarray chunk error name stateful recurrences

## Validation
- `uv run ruff check src/ tests/`
- `uv run ruff format --check src/ tests/`
- `uv run mypy src/`
- `uv run pytest`
- `coderabbit review --agent -t uncommitted` (0 findings)

Benchmark: [#795 comment](https://github.com/monocongo/climate_indices/issues/795#issuecomment-5654158939).

Stacked on #836; targets `feature/794-fire-module-api`, not `main`. Refs #795.

## Summary by Sourcery

Establish the recursive fire-index state and execution contract as the foundation for resumable NumPy implementations and safe xarray integration.

Enhancements:
- Define a shared stateful recurrence contract for fire indices, including resumable state, spin-up behavior, and pure-NumPy execution.
- Add a reusable recurrence engine with validation for time-first inputs, state shapes, complete state transitions, and append-state round trips.
- Require stateful fire xarray adapters to reject Dask inputs split across multiple time chunks while allowing spatial chunking.

Documentation:
- Document the recursive fire state and execution decisions in ADR-0006 and align fire subsystem and xarray compatibility guidance with the contract.

Tests:
- Add recurrence validation, state preservation, zero-length, append round-trip, and benchmark coverage, plus xarray multi-time-chunk error assertions.

## Summary by CodeRabbit

- **Documentation**
  - Documented the planned stateful fire-index contract, including state restoration, continuation across appended data, and spin-up behavior.
  - Clarified that stateful xarray adapters require the full `time` dimension in a single Dask chunk; spatial chunking remains supported.
  - Documented that invalid multi-chunk time inputs raise `CoordinateValidationError` and are not automatically rechunked.
  - Clarified that the weather-only Fosberg index remains the only currently available fire index through the NumPy layer.

- **Bug Fixes**
  - Improved validation messaging to identify stateful recurrences as requiring complete time-series data.

### CODE REVIEW

**Comment by monocongo** [comment]
### 🛠️ Multi-Agent Review Summary
- **Review target:** https://github.com/monocongo/climate_indices/pull/837 (`feature/795-fire-recursive-architecture` → `feature/794-fire-module-api`)
- **Lanes run:** 3 (correctness/edge cases; API/contracts + docs; repository conventions/validation) — all on `kimi-k3` via the project-local `reviewer` lane agent
- **Validation:** `ruff check src/ tests/` pass; `ruff format --check` pass; `mypy src/` pass (19 files); `pytest -q` 1255 passed, 386 deselected; `pytest tests/test_review_scripts.py` 13 passed; llms bundles verified in sync with fresh regeneration
- **Fix commits:** `e5b51a95` fix(fire): own recurrence state and raise argument errors in _recurse; `30ebaa0d` docs(fire): mark stateful recurrence contract as planned API

### Findings and fix status
#### [P3] `_recurse` lets a step function mutate caller input arrays — Fixed
- **Location:** `src/climate_indices/fire.py` (day loop, state rebuild)
- **Origin:** Correctness lane
- **Problem:** state rebuild used `np.asarray(value, dtype=np.float64)`, a no-op for float64. A step returning a view into its forcing input (`return (day,)`) plus an in-place state update writes into the caller's forcing array. Reproduced: `forcing` values changed after `_recurse` returned. Latent today (no production caller yet) but the trap was built into the shared helper that KBDI/CFFWIS will use.
- **Applied fix:** `np.array(value, dtype=np.float64)` so the recurrence owns every state array, plus a regression test with an aliasing, mutating step — commit `e5b51a95`.

#### [P4] Empty `inputs`/`initial_state` raised bare `ValueError` — Fixed
- **Location:** `src/climate_indices/fire.py` (argument guards)
- **Origin:** Correctness lane
- **Problem:** every other complaint in the function raises a library exception (`DataShapeError`); these two raised bare `ValueError`, uncatchable via `ClimateIndicesError`.
- **Applied fix:** raise `InvalidArgumentError` (already the module's taxonomy for bad argument values), with tests — commit `e5b51a95`.

#### [P4] Zero-length time axis behavior unpinned — Fixed
- **Location:** `src/climate_indices/fire.py` (`input_shape[0] == 0` path)
- **Origin:** Correctness lane
- **Problem:** zero-day input silently returns empty values plus the initial state — arguably correct semantics, but no test pinned it.
- **Applied fix:** test asserting `(0, 2)` values and passthrough of the initial state — commit `e5b51a95`.

#### [P4] Design doc describes future APIs in present tense — Fixed
- **Location:** `docs/design/fire-subsystem.md` ("Stateful recurrence contract" section)
- **Origin:** API/contracts + docs lane
- **Problem:** "KBDI, FFMC, DMC, DC, and CFFWIS accept..." plus a runnable-looking `fire.kbdi(...)` example, but none of these APIs exist yet (#799/#803); the neighboring xarray section correctly says "Future".
- **Applied fix:** future-tense framing and an "illustrative of the target #799 API" label on the example — commit `30ebaa0d`.

#### [P4] llms bundles contain `adr/...` links that don't resolve from repo root — Blocked (out of scope)
- **Location:** `llms.txt` / `llms-full.txt` (adr/0006 entry)
- **Origin:** API/contracts + docs lane
- **Problem:** `adr/0006-fire-recursive-state-and-execution.md` doesn't resolve from repo root (actual path `docs/adr/...`). Pre-existing generator behavior — `adr/0004` has the same breakage; the generator copies doc links verbatim. Not introduced by this PR.
- **Blocked:** fixing `scripts/generate_llms_txt.py` to rewrite relative links is a repo-wide change affecting all bundles; recommend a separate issue rather than expanding this PR's scope.

### Remaining blockers
None. All gates pass on final HEAD `30ebaa0d`.

File: src/climate_indices/fire.py [hunk 1]
```diff
@@ -63,6 +64,62 @@
...
+        except ValueError as exc:
+            raise DataShapeError(
+                "Initial recurrence state must broadcast to the spatial input shape.",
+                expected_shape=str(spatial_shape),
+                actual_shape=initial_value.shape,
+            ) from exc
+    state = tuple(state_values)
+    result = np.empty(input_shape, dtype=np.float64)
+    for day in range(input_shape[0]):
+        state = tuple(
```
[the remark this answers is not in the document]
Comment (monocongo) [reply src/climate_indices/fire.py]: Fixed in c1e735f5: `_recurse` now validates the step's returned state — arity must match the initialized tuple (`InvalidArgumentError`) and each state's spatial shape must be preserved (`DataShapeError`). Covered by `test_recurrence_step_must_return_complete_same_shaped_state`.

File: docs/xarray_compatibility.md [hunk 2]
```diff
@@ -21,6 +21,20 @@ may change in a future minor release.
 | Automatic temporal inference | Yes | Monthly and daily time-coordinate inference is covered by adapter tests. |
 | Multi-input alignment | Yes | SPEI aligns precipitation and PET with an inner join and emits a warning when timesteps are dropped. |
 
+## Stateful fire indices
+
+KBDI and CFFWIS moisture-code adapters are recursive: each daily value needs
```
[the remark this answers is not in the document]
Comment (monocongo) [reply docs/xarray_compatibility.md]: Fixed in c1e735f5: the section is retitled "Stateful fire indices (planned)" and now states the adapters are not yet shipped, describing the contract as future work per ADR-0006.
